web/ui/react-app/src/App.tsx

```diff
@@ -5,33 +5,61 @@ import { Container } from 'reactstrap';
 import './App.css';
 import { Router, Redirect } from '@reach/router';
 import { Alerts, Config, Flags, Rules, ServiceDiscovery, Status, Targets, TSDBStatus, PanelList } from './pages';
-import PathPrefixProps from './types/PathPrefixProps';
 
 interface AppProps {
   consolesLink: string | null;
 }
 
-const App: FC<PathPrefixProps & AppProps> = ({ pathPrefix, consolesLink }) => {
+const App: FC<AppProps> = ({ consolesLink }) => {
+  // this dynamically/generically determines the pathPrefix by taking the window location path
+  // and stripping the known endpoint suffixes. works out of the box for both direct hosting
+  // and reverse proxy deployments with no additional configurations required
+  let basePath = window.location.pathname;
+  const paths = [
```

Comment:
This list duplicates https://github.com/prometheus/prometheus/blob/19c190b406c992278aaade63be92ecc7bb6a4921/web/web.go#L70-L83, which incorrectly also contains `/version` (that was never a React UI path, I must have included it by mistake initially).

I think it's ok to have this duplication for now, but in the future we might want to think about passing this list from the Go code to the UI in a similar way as the path prefix was passed before, but instead of replacing multiple independent placeholders in the index.html, we could just render a JSON object in there that contains all config items.